UserApi/UserApi/Helper/UserProfileHelper.cs

```diff
@@ -51,24 +59,23 @@ public async Task<UserProfile> GetUserProfile(string filter)
                 DisplayName = user.DisplayName,
                 FirstName = user.GivenName,
                 LastName = user.Surname,
-                UserRole = userRole.ToString(),
+                UserRole = userRole,
                 CaseType = caseTypes
             };
 
             return response;
         }
 
-        private List<string> GetUserCaseTypes(List<AdGroup> userGroups)
-        {
-            return userGroups.Where(IsCaseType).Select(g => g.ToString()).ToList();
-        }
+        private static bool IsCaseType(AdGroup group) => CaseTypeMappings.ContainsKey(group);
 
-        private bool IsCaseType(AdGroup group)
+        private static IEnumerable<string> GetUserCaseTypes(IEnumerable<AdGroup> userGroups)
         {
-            return group == AdGroup.FinancialRemedy || group == AdGroup.MoneyClaims;
+            return userGroups
+                .Where(IsCaseType)
+                .Select(c => CaseTypeMappings[c]);
         }
 
-        private UserRole GetUserRole(List<AdGroup> userGroups)
+        private static UserRole GetUserRole(ICollection<AdGroup> userGroups)
```

Comment:
Change to IEnumerable as you are not using any of the functionality of ICollection

Reply:
Because I enumerate it several times